### ISSUE TITLE
Make initialisation phase error precise, at least for missing dependencies. 

### ISSUE THREAD

**Issue by hsyed** [GUI/lua-resty-auto-ssl#19]
It would be nice if the modules did some checking at runtime.I just put together a docker image today with no experience with any of the components and so much time was lost round tripping to track down all the things that need to be right. People putting together an alpine distribution will generally not add openssl / curl / bash. 

It would be beneficial to have docker files for a few variants available in the repo even if they aren't published publicly. I'm attaching some of the relevant bits below.

```dockerfile
FROM openresty/openresty:1.9.15.1-alpine

RUN \
    apk add --no-cache --update --repository  http://dl-3.alpinelinux.org/alpine/edge/testing/ --virtual .build-deps \
        luarocks5.2 \
        make \
        gcc \
        libc-dev \
    && apk add openssl curl bash \
    && luarocks-5.2 install lua-resty-auto-ssl \
    && rm -rf ~/.cache \
    && apk del .build-deps \
    && mkdir /etc/resty-auto-ssl

ADD ./docker-entrypoint.sh ./docker-entrypoint.sh

ENTRYPOINT ./docker-entrypoint.sh
```

```sh
#!/bin/sh

cat << EOF > /sbin/start_nginx
  #!/bin/sh
  mkdir -p /var/log/nginx
  ln -sf /dev/stdout /var/log/nginx/access.log
  ln -sf /dev/stderr /var/log/nginx/error.log
  /usr/local/openresty/nginx/sbin/nginx -g "daemon off;" -c /etc/nginx/nginx.conf
EOF

chmod +x /sbin/start_nginx

start_nginx
```

**Comment by GUI** [comment]
Thanks for bringing this up and for the example files. I've added a note to the README about additional system dependencies (https://github.com/GUI/lua-resty-auto-ssl/issues/20), but I like your idea of checking for the dependencies at runtime. That would allow us to provide more helpful error messages. I'll try to add that functionality maybe sometime later this week.

**Comment by GUI** [comment]
There are now dependency checks at install-time and at run-time that should provide better error messages in case any system dependencies (like curl, bash, openssl, etc) are missing. These checks are part of the v0.9.0 release.

Also, thanks for the Alpine docker example. I discovered that I think there was an issue with the `start_sockproc` script when running under alpine, since the the BusyBox version of the `find` command that Alpine uses doesn't support the `-printf` flag. That should also be fixed in the v0.9.0 release.
